src/EPiServer.Marketing.Testing.Web/TestHandler/TestHandler.cs

```diff
@@ -667,7 +667,7 @@ internal void AddProxyEventHandler(IKpi kpi)
 
             // Add the proxyeventhandler only once, if its in our reference counter, just increment
             // the reference.
-            if (!_ReferenceCounter.hasReference(kpi.GetType()))
+            if (_ReferenceCounter.IsNullOrEmpty())
```

Comment:
I think we should use two conditions with AND like this:
if (_ReferenceCounter.IsNullOrEmpty() && !_ReferenceCounter.hasReference(kpi.GetType()))

Reply:
Checking hasReference does not have any usage in this logic. I will make that function obsolete since it is not used anywhere else.